olp-cpp-sdk-authentication/src/AuthenticationClientUtils.cpp

```diff
@@ -262,6 +266,54 @@ AuthorizeResult GetAuthorizeResult(rapidjson::Document& doc) {
   return result;
 }
 
+UserAccountInfoResponse GetUserAccountInfoResponse(rapidjson::Document& doc) {
+  const NameToMethodVector<UserAccountInfoResponse, std::string> string_fields{
+      {Constants::USER_ID, &UserAccountInfoResponse::SetUserId},
+      {Constants::REALM, &UserAccountInfoResponse::SetRealm},
+      {Constants::FACEBOOK_ID, &UserAccountInfoResponse::SetFacebookId},
+      {Constants::FIRSTNAME, &UserAccountInfoResponse::SetFirstname},
+      {Constants::LASTNAME, &UserAccountInfoResponse::SetLastname},
+      {Constants::EMAIL, &UserAccountInfoResponse::SetEmail},
+      {Constants::RECOVERY_EMAIL, &UserAccountInfoResponse::SetRecoveryEmail},
+      {Constants::DOB, &UserAccountInfoResponse::SetDob},
+      {Constants::COUNTRY_CODE, &UserAccountInfoResponse::SetCountryCode},
+      {Constants::LANGUAGE, &UserAccountInfoResponse::SetLanguage},
+      {Constants::PHONE_NUMBER, &UserAccountInfoResponse::SetPhoneNumber},
+      {Constants::STATE, &UserAccountInfoResponse::SetState},
+      {Constants::HRN, &UserAccountInfoResponse::SetHrn},
+      {Constants::ACCOUNT_TYPE, &UserAccountInfoResponse::SetAccountType}};
+
+  const NameToMethodVector<UserAccountInfoResponse, bool> boolean_fields{
+      {Constants::EMAIL_VERIFIED, &UserAccountInfoResponse::SetEmailVerified},
+      {Constants::PHONE_NUMBER_VERIFIED,
+       &UserAccountInfoResponse::SetPhoneNumberVerified},
+      {Constants::MARKETING_ENABLED,
+       &UserAccountInfoResponse::SetMarketingEnabled}};
+
+  const NameToMethodVector<UserAccountInfoResponse, time_t> time_fields{
+      {Constants::CREATED_TIME, &UserAccountInfoResponse::SetCreatedTime},
+      {Constants::UPDATED_TIME, &UserAccountInfoResponse::SetUpdatedTime}};
+
+  UserAccountInfoResponse response;
+
+  for (const auto& string_field : string_fields) {
+    if (doc.HasMember(string_field.first))
+      (response.*string_field.second)(doc[string_field.first].GetString());
```

Comment:
In case this element is not string rapidjson will crash

Reply:
Yes, as every and each call in this file. Do we need to fix all places?